### PR TITLE
Add warn to ensure that system is fully upgraded (RhBug:1225442)

### PR DESCRIPTION
**Pull request by j-mracek**
It should help user to realize that there could be an issue when system is not
fully upgraded.

The patch also enhance the documentation.

https://bugzilla.redhat.com/show_bug.cgi?id=1225442

Requires: https://github.com/rpm-software-management/dnf/pull/841

### CODE REVIEW

File: doc/system-upgrade.rst [hunk 1]
```diff
@@ -24,8 +24,8 @@ Description
 -----------
 
 ``dnf system-upgrade`` can be used to upgrade a Fedora system to a new major
-release. It replaces fedup (the old Fedora Upgrade tool).
-
+release. It replaces fedup (the old Fedora Upgrade tool). Before you proceed ensure that your system
+is fully upgraded (``dnf upgrade --refresh``).
```
Comment (Conan-Kudo) [inline doc/system-upgrade.rst]: It should be `dnf --refresh upgrade`. Technically the order doesn't matter, but that's how the man page displays it, so we should be consistent.

File: plugins/system_upgrade.py [hunk 2]
```diff
@@ -345,6 +345,8 @@ def _call_sub(self, name, *args):
     # == configure_*: set up action-specific demands ==========================
 
     def configure_download(self, *args):
+        msg = _('Please ensure that your system is fully upgraded!')
+        logger.warning(self.base.output.term.bold(msg) + ' ("dnf upgrade --refresh")')
```
Comment (Conan-Kudo) [inline plugins/system_upgrade.py]: Ditto.

File: doc/system-upgrade.rst [hunk 3]
```diff
@@ -111,6 +111,8 @@ Examples
 Typical upgrade usage
 ---------------------
 
+``dnf upgrade --refresh``
```
Comment (Conan-Kudo) [inline doc/system-upgrade.rst]: Ditto.

**Comment by j-mracek** [comment]
I would like to add some commits

**Comment by j-mracek** [comment]
I am going to remove some lines

**Comment by j-mracek** [comment]
There is one disadvantage of the approach presented here, if anyone runs successful  transaction after download, system upgrade will loose cached files and transaction will fail during reboot. 
I have also a plan to remove the upgrade during reboot functionality from system upgrade and provide it to other commands like install, upgrade (requested here https://bugzilla.redhat.com/show_bug.cgi?id=1382063).

File: plugins/snapper.py [hunk 4]
```diff
@@ -18,12 +18,11 @@
 #
 
 from dbus import SystemBus, Interface, DBusException
+from dnfpluginsextras import _, logger
+
 import dnf
-import dnfpluginsextras
 import sys
```
Comment (dmach) [inline plugins/snapper.py]: import order again

File: plugins/system_upgrade.py [hunk 5]
```diff
@@ -20,36 +20,18 @@
...
 from subprocess import call, check_call
+from systemd import journal
 
 import dnf
 import dnf.cli
-from dnf.cli import CliError
-
+import json
+import os
 import uuid
```
Comment (dmach) [inline plugins/system_upgrade.py]: import order again

File: plugins/torproxy.py [hunk 6]
```diff
@@ -17,16 +17,15 @@
 # License and may only be used or replicated with the express permission of
 # Red Hat, Inc.
 #
+from dnfpluginsextras import _, logger
```
Comment (dmach) [inline plugins/torproxy.py]: import order again

File: plugins/tracer.py [hunk 7]
```diff
@@ -21,16 +21,15 @@
 #
 
 from __future__ import absolute_import
+from dnfpluginsextras import _
+from tracer import Query, Package
+from tracer.views.default import DefaultView
```
Comment (dmach) [inline plugins/tracer.py]: import order again

**Comment by j-mracek** [comment]
@dmach Here is an improved version

**Comment by dmach** [comment]
I've reviewed the code and it looks good to me. Merging.

**Pull request merged**